tests/functional/aws-node-sdk/test/multipleBackend/objectCopy.js

```diff
@@ -0,0 +1,178 @@
+        assert.strictEqual(result.ETag, `"${correctMD5}"`);
+        cb();
+    });
+}
+
+function assertGetObjects(sourceKey, sourceBucket, sourceLoc, destKey,
+destBucket, destLoc, awsKey, callback) {
+    const sourceGetParams = { Bucket: sourceBucket, Key: sourceKey };
+    const destGetParams = { Bucket: destBucket, Key: destKey };
+    const awsParams = { Bucket: awsBucket, Key: awsKey };
+
+    async.series([
+        cb => s3.getObject(sourceGetParams, cb),
+        cb => s3.getObject(destGetParams, cb),
+        cb => awsS3.getObject(awsParams, cb),
+    ], (err, results) => {
+        assert.equal(err, null, `Error in assertGetObjects: ${err}`);
+
+        const [sourceRes, destRes, awsRes] = results;
+        if (process.env.ENABLE_KMS_ENCRYPTION === 'true') {
+            assert.strictEqual(sourceRes.ServerSideEncryption, 'AES256');
+            assert.strictEqual(destRes.ServerSideEncryption, 'AES256');
+            assert.strictEqual(awsRes.ServerSideEncryption, 'AES256');
+        } else {
+            assert.strictEqual(sourceRes.ETag, `"${correctMD5}"`);
+            assert.strictEqual(destRes.ETag, `"${correctMD5}"`);
+            assert.deepStrictEqual(sourceRes.Body, destRes.Body);
+            assert.strictEqual(awsRes.ETag, `"${correctMD5}"`);
+            assert.deepStrictEqual(sourceRes.Body, awsRes.Body);
+        }
+        assert.strictEqual(sourceRes.Metadata[locMetaHeader], sourceLoc);
+        assert.strictEqual(destRes.Metadata[locMetaHeader], destLoc);
+        callback();
+    });
+}
+
+describeSkipIfNotMultiple('MultipleBackend object copy', function testSuite() {
+    this.timeout(250000);
+    withV4(sigCfg => {
+        beforeEach(() => {
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            const awsConfig = getRealAwsConfig(awsLocation);
+            awsS3 = new AWS.S3(awsConfig);
+            process.stdout.write('Creating bucket\n');
+            if (process.env.ENABLE_KMS_ENCRYPTION === 'true') {
+                s3.createBucketAsync = createEncryptedBucketPromise;
+            }
+            return s3.createBucketAsync({ Bucket: bucket })
+            .catch(err => {
+                process.stdout.write(`Error creating bucket: ${err}\n`);
+                throw err;
+            });
+        });
+
+        afterEach(() => {
+            process.stdout.write('Emptying bucket\n');
+            return bucketUtil.empty(bucket)
+            .then(() => {
+                process.stdout.write('Deleting bucket\n');
+                return bucketUtil.deleteOne(bucket);
+            })
+            .catch(err => {
+                process.stdout.write(`Error in afterEach: ${err}\n`);
+                throw err;
+            });
+        });
+
+        it('should copy an object from mem to AWS', done => {
+            putSourceObj('mem', () => {
+                const copyParams = Object.assign({
+                    MetadataDirective: 'REPLACE',
```

Comment:
Thanks. As discussed, if someone is copying from location constraint x to location constraint y and the source object metadata has a location header (scal-location-constraint) and they have selected the "COPY" option for metadata, we should copy all of the metadata except for the location header (scal-location-constraint) which we should have as the actual destination location.